src/web/rustdoc.rs

```diff
@@ -795,6 +798,29 @@ impl Handler for SharedResourceHandler {
     }
 }
 
+/// Serves shared web resources used by rustdoc-generated documentation.
+///
+/// Rustdoc has certain JS, CSS, font and image files that are required for all
+/// documentation it generates, and these don't change often. We make one copy
+/// of these per rustdoc release and serve them from a common location.
+///
+/// This handler considers the whole path, and looks for a file at that path in
+/// the Storage.
+pub struct SharedResourceHandler;
```

Comment:
nit, not a blocker: most handlers are defined as functions, this could also be one.